pydgraph/client.py

```diff
@@ -61,35 +61,41 @@ def check_version(self, timeout=None, metadata=None, credentials=None):
                                                    credentials=credentials)
                 return response.tag
             else:
-                raise error
+                self._common_except(error)
 
     def login(self, userid, password, timeout=None, metadata=None,
               credentials=None):
-        login_req = api.LoginRequest()
-        login_req.userid = userid
-        login_req.password = password
-        login_req.namespace = 0
-
-        response = self.any_client().login(login_req, timeout=timeout,
-                                           metadata=metadata,
-                                           credentials=credentials)
-        self._jwt = api.Jwt()
-        self._jwt.ParseFromString(response.json)
-        self._login_metadata = [("accessjwt", self._jwt.access_jwt)]
+        try:
+            login_req = api.LoginRequest()
```

Comment:
I think we could write a decorator to do this instead of copy pasting the same code in all the functions. Can you check if that is doable?